debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -1012,6 +1013,28 @@ public void readSchema(Tables tables, String databaseCatalog, String schemaNameP
         return jdbcTypeByLocalTypeName;
     }
 
+    private void cleanupPreparedStatement(PreparedStatement statement) {
+        LOGGER.trace("Closing prepared statement '{}' removed from cache", statement);
+        try {
+            statement.close();
+        }
+        catch (Exception e) {
+            LOGGER.info("Exception while closing a prepared statement removed from cache", e);
+        }
+    }
+
+    private PreparedStatement createPreparedStatement(String preparedQueryString) {
+        return statementCache.computeIfAbsent(preparedQueryString, query -> {
```

Comment:
As per Ben's comment on the other PR, that operation actually isn't atomic. I think here it's doing some harm, as we may end up with dangling unclosed statements.

Reply:
For the sake of closure, `JdbcConnection` is not meant to be accessed concurrently, so this usage should be fine.